python-summary-statistics/statistics.py

```diff
@@ -76,5 +164,39 @@ def update_schema_field(fields, name, value):
             field['type'] = value
 
 
+def get_X(dep_var, indep_vars):
+    """Create dataframe from input data.
+    :param dep_var:
+    :param indep_vars:
+    :return: dataframe with data from all variables
+
+    TODO: move this function to `io_helper` and reuse it in python-sgd-regressor
+    """
+    df = {}
+    for var in [dep_var] + indep_vars:
+        # categorical variable - we need to add all categories to make one-hot encoding work right
+        if is_nominal(var['type']['name']):
+            df[var['name']] = pd.Categorical(var['series'], categories=var['type']['enumeration'])
+        else:
+            # infer type automatically
+            df[var['name']] = var['series']
+    X = pd.DataFrame(df)
+    return X
+
+
 if __name__ == '__main__':
-    main()
+    parser = argparse.ArgumentParser()
+    parser.add_argument('compute', choices=['compute'])
+    parser.add_argument('--mode', choices=['intermediate', 'aggregate'], default='intermediate')
+    # QUESTION: (job_id, node) is a primary key of `job_result` table. Does it mean I'll need node ids as well in order
+    # to query unique job?
```

Comment:
(job_id, node) is a primary key of `job_result` table. Does it mean I'll need node ids as well in order to query unique job?